kv/kvclient/kvcoord: TestProxyTracing failed

kv/kvclient/kvcoord.TestProxyTracing [failed](https://tanzanite.cluster.engflow.com/invocations/default/358a8882-1c64-4815-bd52-33736a313798?testReportRun=3&testReportShard=10&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZjbGllbnQva3Zjb29yZDprdmNvb3JkX3Rlc3Q=) on release-24.3.0-rc @ [49d902f212d10c8482298d777061ba153d5c16c9](https://github.com/cockroachdb/cockroach/commits/49d902f212d10c8482298d777061ba153d5c16c9):


```
=== RUN   TestProxyTracing
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestProxyTracing3709837767
    test_log_scope.go:76: use -show-logs to present logs inline
    dist_sender_server_test.go:4943: -- test log scope end --
test logs left over in: outputs.zip/logTestProxyTracing3709837767
--- FAIL: TestProxyTracing (455.30s)
=== RUN   TestProxyTracing/lease-type=LeaseExpiration
    test_server_shim.go:139: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    dist_sender_server_test.go:4937: request succeeded without proxying
    --- FAIL: TestProxyTracing/lease-type=LeaseExpiration (237.80s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=10</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #134952 kv/kvclient/kvcoord: TestProxyTracing failed [C-bug C-test-failure O-robot P-2 T-kv branch-release-24.3]
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestProxyTracing.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-44594

--- a/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go
+++ b/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go
@@ -4836,6 +4836,9 @@ func TestProxyTracing(t *testing.T) {
 		kvserver.OverrideDefaultLeaseType(ctx, &st.SV, leaseType)
 		kvserver.RangefeedEnabled.Override(ctx, &st.SV, true)
 		kvserver.RangeFeedRefreshInterval.Override(ctx, &st.SV, 10*time.Millisecond)
+		// Disable follower reads to ensure that the request is proxied, and not
+		// answered locally due to follower reads.
+		kvserver.FollowerReadsEnabled.Override(ctx, &st.SV, false)
 		closedts.TargetDuration.Override(ctx, &st.SV, 10*time.Millisecond)
 		closedts.SideTransportCloseInterval.Override(ctx, &st.SV, 10*time.Millisecond)
 

